peerpod-ctrl/Dockerfile

```diff
@@ -1,12 +1,12 @@
 # Build the manager binary
-FROM quay.io/confidential-containers/golang-fedora:1.20.7-36 as builder
+FROM --platform=$TARGETPLATFORM quay.io/confidential-containers/golang-fedora:1.20.8-36 as builder
```

Comment:
Should this be `BUILDPLATFORM` instead?

Reply:
I think we use the same arch for builder when build the binaries, so $TARGETPLATFORM is right here? Which is linux/s390x for s390x.

Reply:
Using `BUILDPLATFORM` for builder, error like this:  https://github.com/confidential-containers/cloud-api-adaptor/actions/runs/6192335422/job/16812236353#step:6:732

Reply:
I guess amd64 golang-fedora image does not have CGO compiler to compile s390x and ppc64le, so we need use corresponding arch image as builder to use CGO, added `CC` arg to avoid the symbol link.

Reply:
I think the GHA against above commits https://github.com/confidential-containers/cloud-api-adaptor/actions/workflows/peerpod-ctrl_image.yaml shows why we need use `TARGETPLATFORM`. Maybe there is alternatives, e.g. built all archs cgo compilers in the builder image -- `golang-fedora`?

Reply:
To do the cross complication you would also need to turn off cgo. But if this way works too then that's fine.

Reply:
@jtumber-ibm So I guess you're OK with the change? Would you  please approve the PR if there is no more concerns?